tools/wpt/browser.py

```diff
@@ -111,27 +92,46 @@ def install(self, dest=None):
         from mozdownload import FactoryScraper
         import mozinstall
 
+        platform = {
+            "Linux": "linux",
+            "Windows": "win",
+            "Darwin": "mac"
+        }.get(uname[0])
+
+        if platform is None:
+            # This was earlier checked in platform_string, which isn't needed anymore since we switched
+            # to using mozdownload.
+            raise ValueError("Unable to construct a valid Firefox package name for current platform")
+
         if dest is None:
             dest = os.getcwd()
 
         filename = FactoryScraper('daily', branch='mozilla-central', destination=dest).download()
 
         try:
-            mozinstall.install(filename, dest)
+            install_dest = mozinstall.install(filename, dest)
         except mozinstall.mozinstall.InstallError as e:
-            if uname[0] == "Darwin":
+            if platform == "mac":
                 # mozinstall will fail here if nightly is already installed in the venv
                 # This only occurs on macOS because shutil.copy_tree() is called in
                 # mozinstall._install_dmg and will fail if the file already exists.
                 # copytree isn't used while installing on Windows/linux, so the same error
                 # won't be thrown if we try to rewrite there.
                 mozinstall.uninstall(dest+'/Firefox Nightly.app')
-                mozinstall.install(filename, dest)
+                install_dest = mozinstall.install(filename, dest)
             else:
                 raise
 
         os.remove(filename)
-        return find_executable("firefox", os.path.join(dest, "firefox"))
+
+        if platform == "linux":
+            bin_path = os.path.join(install_dest, "firefox")
+        elif platform == "win":
```

Comment:
I wonder if we should add `firefox` to the `dest` in non-linux cases, to ensure we aren't overwriting existing files.

Reply:
Afaik it shouldn't overwrite any files besides the firefox ones in non-linux cases. 
On mac, it installs the .dmg to make the firefox nightly app, and in windows, the installer sets it up so that the firefox build is within a new folder named with the nightly version and language it installed. 
Not sure if these can lead to us overwriting anything, but then again I don't have much experience with the two OSes.